### PR TITLE
Add Diffblue tests

### PR DESCRIPTION
**Pull request by sudo-jaa**
Current coverage - 36.53%
Increase - 12.54%
3015 tests generated. 

Here is a selection of interesting tests from Diffblue Deeptest upon analysis of Activiti.

### CODE REVIEW

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 82 branch to line 82
+   *  - conditional line 82 branch to line 86
+   *  - conditional line 86 branch to line 86
+   */
+
+  @Test
+  public void isBitOnInputPositivePositiveOutputTrue() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 82 branch to line 82
+   *  - conditional line 82 branch to line 86
+   *  - conditional line 86 branch to line 86
+   */
+
+  @Test
+  public void isBitOnInputZeroPositiveOutputFalse() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 100 branch to line 103
+   *  - conditional line 103 branch to line 103
+   */
+
+  @Test
+  public void setBitInputZeroPositiveFalseOutputZero() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 100 branch to line 101
+   *  - conditional line 101 branch to line 101
+   */
+
+  @Test
+  public void setBitInputZeroPositiveTrueOutputPositive() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 65 branch to line 65
+   *  - conditional line 65 branch to line 70
+   */
+
+  @Test
+  public void setBitOffInputZeroPositiveOutputZero() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 48 branch to line 48
+   *  - conditional line 48 branch to line 53
+   */
+
+  @Test
+  public void setBitOnInputPositivePositiveOutputPositive() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+   *     - conditional line 50 branch to line 50
+   *     - conditional line 51 branch to line 51
+   *     - conditional line 51 branch to line 52
+   *     - conditional line 53 branch to line 53
+   *     - conditional line 54 branch to line 54
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:44]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+   *     - conditional line 50 branch to line 50
+   *     - conditional line 51 branch to line 51
+   *     - conditional line 51 branch to line 52
+   *     - conditional line 53 branch to line 53
+   *     - conditional line 54 branch to line 54
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull4() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:69]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+   *  - org/activiti/engine/impl/util/json/Cookie.java:49: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 50 branch to line 50
+   *     - conditional line 51 branch to line 51
+   *     - conditional line 51 branch to line 56
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull3() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:92]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/Cookie.java:141: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 142 branch to line 142
+   *     - conditional line 143 branch to line 144
+   */
+
+  @Test
+  public void unescapeInputNotNullOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:114]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+   *     - conditional line 145 branch to line 146
+   *     - conditional line 146 branch to line 146
+   *     - conditional line 147 branch to line 147
+   *     - conditional line 148 branch to line 148
+   *     - conditional line 148 branch to line 149
+   */
+
+  @Test
+  public void unescapeInputNotNullOutputNotNull3() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:142]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 1024 branch to line 1024
+   *  - conditional line 1024 branch to line 1025
+   */
+
+  @Test
+  public void quoteInputNotNullOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java:39]: :+1: It would have been nicer had it put something other than an empty string in, but saying that, this could be an edge case, which is nice.

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+   *  - conditional line 1126 branch to line 1127
+   *  - conditional line 1127 branch to line 1134
+   *  - conditional line 1134 branch to line 1134
+   *  - conditional line 1134 branch to line 1137
+   *  - conditional line 1138 branch to line 1139
+   */
+
+  @Test
+  public void stringToValueInputNotNullOutputNegative() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java:68]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+   *  - conditional line 1126 branch to line 1127
+   *  - conditional line 1127 branch to line 1134
+   *  - conditional line 1134 branch to line 1134
+   *  - conditional line 1134 branch to line 1137
+   *  - conditional line 1138 branch to line 1139
+   */
+
+  @Test
+  public void stringToValueInputNotNullOutputPositive() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java:97]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+   *  - conditional line 1127 branch to line 1127
+   *  - conditional line 1127 branch to line 1134
+   *  - conditional line 1134 branch to line 1134
+   *  - conditional line 1134 branch to line 1137
+   *  - conditional line 1138 branch to line 1139
+   */
+
+  @Test
+  public void stringToValueInputNotNullOutputPositive3() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java:127]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+   *  - conditional line 1126 branch to line 1127
+   *  - conditional line 1127 branch to line 1134
+   *  - conditional line 1134 branch to line 1134
+   *  - conditional line 1134 branch to line 1137
+   *  - conditional line 1138 branch to line 1141
+   */
+
+  @Test
+  public void stringToValueInputNotNullOutputPositive4() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONObjectTest.java:156]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 95 branch to line 95
+   *  - conditional line 95 branch to line 96
+   */
+
+  @Test
+  public void dehexcharInput0OutputZero() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java:39]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+   * This test case covers:
+   *  - conditional line 95 branch to line 95
+   *  - conditional line 95 branch to line 98
+   *  - conditional line 98 branch to line 98
+   *  - conditional line 98 branch to line 99
+   */
+
+  @Test
+  public void dehexcharInputFOutputPositive() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java:85]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+   *  - conditional line 95 branch to line 95
+   *  - conditional line 95 branch to line 98
+   *  - conditional line 98 branch to line 98
+   *  - conditional line 98 branch to line 101
+   *  - conditional line 101 branch to line 104
+   */
+
+  @Test
+  public void dehexcharInputGOutputNegative() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java:108]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 95 branch to line 98
+   *  - conditional line 98 branch to line 101
+   *  - conditional line 101 branch to line 104
+   */
+
+  @Test
+  public void dehexcharInputNotNullOutputNegative() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java:129]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+   *  - conditional line 95 branch to line 98
+   *  - conditional line 98 branch to line 98
+   *  - conditional line 98 branch to line 101
+   *  - conditional line 101 branch to line 101
+   *  - conditional line 101 branch to line 104
+   */
+
+  @Test
+  public void dehexcharInputoOutputNegative() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/JSONTokenerTest.java:153]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/XML.java:80: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 81 branch to line 81
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java:40]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java [hunk 23]
```diff
@@ -0,0 +1,122 @@
...
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/XML.java:80: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 81 branch to line 81
+   *     - case 3 of switch on line 82
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull3() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java:62]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/XML.java:80: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 81 branch to line 81
+   *     - case 1 of switch on line 82
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull4() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java:84]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java [hunk 25]
```diff
@@ -0,0 +1,122 @@
...
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/XML.java:80: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 81 branch to line 81
+   *     - case 4 of switch on line 82
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull6() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java:106]: :+1:

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+   * This test case covers:
+   *  - org/activiti/engine/impl/util/json/XML.java:80: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 81 branch to line 81
+   *     - case 2 of switch on line 82
+   */
+
+  @Test
+  public void escapeInputNotNullOutputNotNull5() {
+
```
Comment (JohnLBergqvist) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/XMLTest.java:128]: :+1:

**Review by JohnLBergqvist — APPROVED** [review]
These tests are Gawjus! :kiss:

**Review by petersear — APPROVED** [review]
Impressive!
I didn't mark individual tests as they all look good to me.

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java [hunk 27]
```diff
@@ -0,0 +1,143 @@
+package org.activiti.engine.test.impl.util.json;
+
+import org.activiti.engine.impl.util.BitMaskUtil;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BitMaskUtilTest {
```
Comment (paulkennethkent) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/BitMaskUtilTest.java]: Couldn't find the source so the test is possibly in the wrong folder.

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java (hunk 11 above)
Comment (paulkennethkent) [reply activiti-engine/src/test/java/org/activiti/engine/test/impl/util/json/CookieTest.java:142]: 👍

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/BitMaskUtilTest.java [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+    // Arrange
+    final int value = 128;
+    final int bitNumber = 6;
+
+    // Act
+    final int retval = BitMaskUtil.setBitOn(value, bitNumber);
+
+    // Assert result
+    Assert.assertEquals(160, retval);
+  }
```
Comment (paulkennethkent) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/BitMaskUtilTest.java:160]: All good 👍 - do we have the case where it throws the exception or is that already covered?

**Review by peterschrammel — APPROVED** [review]
All tests excellent!!!

**Comment by paulkennethkent** [comment]
Make sure the final commit is done by @diffblue-assistant and get that PR in.

File: activiti-engine/src/test/java/org/activiti/engine/test/impl/util/BitMaskUtilTest.java [hunk 29]
```diff
@@ -0,0 +1,249 @@
...
+    final int value = 160;
+    final int bitNumber = -2_147_483_642;
+
+    // Act
+    thrown.expect(IllegalArgumentException.class);
+    BitMaskUtil.setBitOn(value, bitNumber);
+
+    // Method is not expected to return due to exception thrown
+  }
+}
```
Comment (peterschrammel) [inline activiti-engine/src/test/java/org/activiti/engine/test/impl/util/BitMaskUtilTest.java]: Missing newline

**Review by paulkennethkent — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]
